ingredient/ingredient-postgresql/src/plugin.ts

```diff
@@ -1,28 +1,143 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { BaseIngredient, IngredientManager,IIngredient,DeploymentContext } from "@azbake/core"
 import { ARMHelper } from "@azbake/arm-helper"
-import ARMTemplate from "./FlexibleServerArm.json" 
+import { PostgreSQLDBUtils } from "./functions"
+import { VnetData } from "./vnetData"
+import PublicAccessARMTemplate from "./PublicAccessArm.json" 
+import PrivateAccessARMTemplate from "./PrivateAccessArm.json"
+import { Subnet,VirtualNetwork } from "@azure/arm-network/esm/models"
 
 export class PostgreSQLDB extends BaseIngredient {
 
     constructor(name: string, ingredient: IIngredient, ctx: DeploymentContext) {
         super(name, ingredient, ctx);
         this._helper = new ARMHelper(this._ctx);
+        this._functions = new PostgreSQLDBUtils(this._ctx);
+        this._access = this._ingredient.properties.parameters.get("access")?._value.toLowerCase();
```

Comment:
You shouldn't try and access properties in the ctor. Should only do that at execute time, since the execution context can change for different regions/etc.

Reply:
You should also not directly access param values, and instead resolve them correctly.. since access could be an expression.

Reply:
ah, I see, thanks. I moved the `this._access` assignment to the top of `Execute()` so that it can use the parameters from `BakeParamsToARMParamsAsync()` which I believe fixes this